### ISSUE TITLE
UI component definition.map.xml overlooks XSD-valid nodes

### ISSUE THREAD

**Issue by neillrobson** [magento/magento2#14140]
### Preconditions

The issue was discovered in Magento 2.2.2 EE, but looks like it still exists in the 2.2-develop branch of this repository. 

### Steps to reproduce

1. Create and configure a new UI form component with the following XML definition:

```
<form xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
  <!-- Include standard <argument>, <settings>, <dataSource> to create minimal form. -->
  <fieldset name="billing">
    <field name="shippingMethod" formElement="select">
      <formElements>
        <select>
          <settings>
            <options class="Magento\Shipping\Model\Config\Source\Allmethods" />
            <label>Shipping Method</label>
          </settings>
        </select>
      </formElements>
    </field>
  </fieldset>
</form>
```

### Expected result

1. The XML should validate with the provided XSD files. (It does.)
2. When viewing the form in the browser, a dropdown select box with the label "Shipping Method" should be shown and populated with the values defined in the `Allmethods` class.

**OR**

1. The XML should not validate with the provided XSD files, because the parser will not read the configuration correctly.

### Actual result

1. The XML validates.
2. The dropdown select box is populated correctly, but the label is not shown.

However, the following code does display the label in the generated markup:

```
<form xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
  <!-- Include standard <argument>, <settings>, <dataSource> to create minimal form. -->
  <fieldset name="billing">
    <field name="shippingMethod" formElement="select">
      <settings>
        <label>Shipping Method</label>
      </settings>
      <formElements>
        <select>
          <settings>
            <options class="Magento\Shipping\Model\Config\Source\Allmethods" />
          </settings>
        </select>
      </formElements>
    </field>
  </fieldset>
</form>
```

The issue seems to be how `definition.map.xml` is written, specifically around [line 285](https://github.com/magento/magento2/blob/cb0954add41284de1b003fa378551b3e973b306c/app/code/Magento/Ui/view/base/ui_component/etc/definition.map.xml#L285) and [787](https://github.com/magento/magento2/blob/cb0954add41284de1b003fa378551b3e973b306c/app/code/Magento/Ui/view/base/ui_component/etc/definition.map.xml#L787). Settings such as `<label>`, which are common to many UI components, are read only from a top-level `<settings>` node, while component-specific settings such as `<options>` are read from the appropriate `<settings>` descendent of `<formElements>`. 

This constraint/decision is perfectly understandable, _but_ it is not represented in the corresponding XSD (see [line 271](https://github.com/magento/magento2/blob/cb0954add41284de1b003fa378551b3e973b306c/app/code/Magento/Ui/etc/ui_configuration.xsd#L271) of ui_configuration.xsd, and the corresponding `formElements` group). Any child from the `abstractSettings` group (found [here](https://github.com/magento/magento2/blob/cb0954add41284de1b003fa378551b3e973b306c/app/code/Magento/Ui/view/base/ui_component/etc/definition/ui_settings.xsd#L866)) is allowed (valid) within the `<settings>` nodes that are descendents of `<formElements>`, but none of them are read by the configuration reader because they aren't specified in definition.map.xml.

### Potential Solutions

Depending on the architectural desires of the Magento team, this miscommunication could be fixed in a few ways:

1. Adding the appropriate xpath mappings for `abstractSettings` within every component-specific settings mapping.
2. Clarifying/constraining the XSD files to ensure that `abstractSettings` are only placed in top-level `<settings>` nodes, and component-specific settings remain isolated.

In short, the new UI component semantics function properly, but permit certain undefined behaviors that should be clarified in both the code itself and in documentation.

**Comment by VladimirZaets** [comment]
Hi, @RNanoware. 
The better solution for fix this issue is to change XSD schema for validating cases when common fields properties are placed inside `formElement`. 

Firstly, we should have the same lists of elements that we can use as `formElement` because for now, they are different.

[formElements list inside ui_configuration.xsd](https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Ui/etc/ui_configuration.xsd#L15) and [formElements list inside ui_settings.xsd](https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Ui/view/base/ui_component/etc/definition/ui_settings.xsd#L49)

The valid list is in the  ui_settings.xsd, so we should delete redundant elements from  ui_configuration.

**Comment by ishakhsuvarov** [comment]
Hi @RNanoware 
Would you like to submit a Pull Request with a fix for this problem? It will speed up delivery in the scope of future patch releases.

**Comment by neillrobson** [comment]
I would certainly be glad to put together a patch for this issue. I may not get around to it immediately due to my schedule, so if anyone else wants to grab it, they can also feel free!

**Comment by ishakhsuvarov** [comment]
@RNanoware Thanks!

**Comment by magento-engcom-team** [comment]
@RNanoware, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by VladimirZaets** [comment]
Hi @RNanoware. Thank you for your report. 
The issue has been fixed in magento/magento2#15161 by @RNanoware in 2.3-develop branch
Related commit(s):
 - [2d75f2522110d4bdf356b73145c2615596bca970](https://github.com/magento/magento2/commit/2d75f2522110d4bdf356b73145c2615596bca970)
 - [b32a33a804db93caf2d61a2b4bc69f9ca9a1d8b4](https://github.com/magento/magento2/commit/b32a33a804db93caf2d61a2b4bc69f9ca9a1d8b4)
 - [56c760950d7b83bb7e4f52cf5624add0e6e3a659](https://github.com/magento/magento2/commit/56c760950d7b83bb7e4f52cf5624add0e6e3a659)
 - [64adba9cc85aafe0050fb2aa7559d520111f759b](https://github.com/magento/magento2/commit/64adba9cc85aafe0050fb2aa7559d520111f759b)
 - [0750a167163e99d65facf8a6acfefd7c059561de](https://github.com/magento/magento2/commit/0750a167163e99d65facf8a6acfefd7c059561de)
 - [214ae8b8ae7a3dac476310dee802561b8039d5ea](https://github.com/magento/magento2/commit/214ae8b8ae7a3dac476310dee802561b8039d5ea)
 - [5a323af50295948925e3bce0c286c085d471ddb0](https://github.com/magento/magento2/commit/5a323af50295948925e3bce0c286c085d471ddb0)

The fix will be available with the upcoming 2.3.0 release.

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by TomashKhamlai** [comment]
The issue was re-tested and we can confirm that it was fixed on the 2.3 release branch. We closing this issue as fixed due to upcoming 2.3 release that will be available soon.
